### PR TITLE
Refactor LS package loader to load modules

### LINKED ISSUE #40883: [Bug]: `import` does not suggest the sub modules
### Description

When we try to import the sub-modules of an external module, we don't get any suggestion after typing the import line.

### Steps to Reproduce

See the recording below:

https://github.com/ballerina-platform/ballerina-lang/assets/63336800/8adfe519-e63d-43d2-9f6b-d7afcc96fcef

### Affected Version(s)

_No response_

### OS, DB, other environment details and versions

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by malinthar**
The LS package loader was only aware of the packages in its environment(distribution, remote and local repositories) before this PR. With this we have added the capability to the LS Package loader to load modules. This change  introduces several improvements to the LS User experience. 

1. Import completion items and code actions for all the modules of a package. 
2. Service Templates for Listener symbols in all the modules

Further, this PR cleans up the LSPackage loader and ServiceTemplateGenerator code base. 

Fixes #40883 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/LSPackageLoader.java [hunk 1]
```diff
@@ -15,186 +15,311 @@
...
+                languageClient.createProgress(workDoneProgressCreateParams);
+
+                // Start progress
+                WorkDoneProgressBegin beginNotification = new WorkDoneProgressBegin();
+                beginNotification.setTitle("Package Loader");
+                beginNotification.setCancellable(false);
+                beginNotification.setMessage("Initializing...");
+                languageClient.notifyProgress(new ProgressParams(Either.forLeft(taskId),
+                        Either.forLeft(beginNotification)));
+            }
```
Comment (dulajdilshan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/LSPackageLoader.java:158]: can flip the `if` condition 
```
if (languageClient == null) {
    return;
}
// Rest of the logic
```

File: language-server/modules/langserver-core/src/test/resources/lspackageloader/user-home/repositories/local/bala/lstest/ls_local1/0.1.0/any/dependency-graph.json [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    },
+    {
+      "org": "lstest",
+      "package_name": "ls_local1",
+      "version": "0.1.0",
+      "module_name": "ls_local1.module1",
+      "dependencies": []
+    }
+  ]
+}
```
Comment (nipunayf) [inline language-server/modules/langserver-core/src/test/resources/lspackageloader/user-home/repositories/local/bala/lstest/ls_local1/0.1.0/any/dependency-graph.json:109]: There are some cases like this that require a newline at the bottom.

Comment (malinthar) [reply language-server/modules/langserver-core/src/test/resources/lspackageloader/user-home/repositories/local/bala/lstest/ls_local1/0.1.0/any/dependency-graph.json:109]: These are generated(bala) files in .ballerina. We can consider those as ballerina archive files. Hense, adding newlines is not required as it is not a source code.

**Comment by dulajdilshan** [comment]
Closing this since we are not planning to do this change recently.

**Pull request closed without merging**